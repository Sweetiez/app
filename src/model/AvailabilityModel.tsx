export default class AvailabilityModel {
  totalPlaces: number;
  placeTaken: number;
}
