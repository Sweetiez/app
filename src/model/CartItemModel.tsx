import ProductCardModel from './ProductCardModel';

export default class CartModel {
  item: ProductCardModel | undefined;
  quantity: number | undefined;
}
