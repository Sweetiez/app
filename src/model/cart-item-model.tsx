import ProductCardModel from './product-card-model';

export default class CartModel {
  item: ProductCardModel | undefined;
  quantity: number | undefined;
}
