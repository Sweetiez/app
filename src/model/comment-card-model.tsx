export default class CommentCardModel {
  id: string | undefined;
  author: string | undefined;
  content: string | undefined;
  rating: number | undefined;
  date: string | undefined;
  authorName: string | undefined;
}
