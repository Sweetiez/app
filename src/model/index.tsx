export {default as LoginRequest} from './LoginRequestModel';
export {default as LoginResponse} from './LoginResponseModel';
export {default as Cart} from './CartItemModel';
export {default as CommentCard} from './CommentCardModel';
export {default as ProductCard} from './ProductCardModel';
export {default as Recipe} from './RecipesModel';
export {default as RecipeStep} from './RecipeStepModel';
export {default as Store} from './StoreModel';
export {default as RegisterRequest} from './RegisterRequestModel';
export {default as UpdateRequest} from './UpdateRequestModel';
export {default as UpdatePasswordRequest} from './UpdatePasswordRequestModel';
export {default as CommentRequest} from './CommentRequestModel';
export {default as User} from './UserModel';
export {default as Evaluation} from './EvaluationModel';
export {default as ReportEvaluationRequest} from './ReportEvaluationModel';
export {default as Order} from './OrderModel';
export {default as OrderedProduct} from './OrderedProductModel';
export {default as FilterModel} from './FilterModel';
export {default as RewardModel} from './RewardModel';
export {default as ResetPasswordRequest} from './ResetPasswordRequestModel';
