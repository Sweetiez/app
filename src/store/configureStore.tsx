import {createStore, combineReducers} from 'redux';
import cartReducer from './reducers/cart';
import shopReducer from './reducers/shop';
import userReducer from './reducers/user';
import ordersReducer from './reducers/orders';
import recipesReducer from './reducers/recipes';

const rootReducer = combineReducers({
  cart: cartReducer,
  shop: shopReducer,
  user: userReducer,
  orders: ordersReducer,
  recipes: recipesReducer,
});
const confStore = () => {
  return createStore(rootReducer);
};
export default confStore;
