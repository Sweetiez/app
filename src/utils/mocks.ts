jest.mock('./../assets/icons/pen.svg', () => 'Pen');
jest.mock('./../assets/icons/star.svg', () => 'Star');
jest.mock('./../assets/icons/plus.svg', () => 'Plus');
jest.mock('./../assets/icons/minus.svg', () => 'Minus');
jest.mock('./../assets/icons/add.svg', () => 'Add');
jest.mock('./../assets/icons/arrow-back.svg', () => 'Back');
jest.mock('./../assets/icons/cook-book.svg', () => 'CookBook');
jest.mock('./../assets/icons/cake.svg', () => 'Cake');
jest.mock('./../assets/icons/event.svg', () => 'Event');
jest.mock('./../assets/icons/eye.svg', () => 'Eye');
jest.mock('./../assets/icons/trash.svg', () => 'Trash');
jest.mock('./../assets/icons/no-image.svg', () => 'NoImage');
jest.mock('./../assets/icons/edit.svg', () => 'Edit');
jest.mock('./../assets/icons/warning.svg', () => 'Warning');

jest.mock('react-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}))



jest.mock('react-native-picker-select', () => 'RNPickerSelect')
jest.mock('react-native-config', () => 'Config')