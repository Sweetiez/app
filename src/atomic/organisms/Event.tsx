import * as React from 'react';
import styled from 'styled-components';
import {useTranslation} from 'react-i18next';
import {Text as SpecialText, Button} from '../atoms';
import {EventModel} from '../../model';
import {formatDate, formatHourDate} from '../../utils/date';
import colors from '../../assets/colors';
import {useState} from 'react';
import {getPublishedEvents, registerToEvents} from '../../store/api/event';
import {EVENTS_ERROR, SUBSCRIBE_ERROR} from '../../store/constants';
import {updateEvents} from '../../store/actions/event';
import {useDispatch} from 'react-redux';

interface Props {
  event: EventModel;
  token?: string;
  userId?: string;
}

const Error = styled.Text`
  color: ${colors.red};
  font-size: 12px;
  margin-right: auto;
  margin-left: auto;
`;
const Content = styled.View`
  flex: 1;
`;
const Container = styled.View`
  flex-direction: row;
  border-radius: 10px;
  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.07);
  background-color: white;
  margin: 5px;
  align-items: center;
  padding: 5px;
`;
const Space = styled.View`
  height: 10px;
`;
const SmallSpace = styled.View`
  height: 5px;
`;
const Messages = styled.View`
  margin-bottom: 5px;
`;

const SubscribeButton = styled.View`
  position: absolute;
  left: 0;
  bottom: 0;
`;

const Event: React.FC<Props> = ({event, token, userId}) => {
  const {t} = useTranslation();
  const dispatch = useDispatch();
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const [error, setError] = useState<boolean>(false);

  const subscribe = () => {
    setIsLoading(true);
    const request = {eventId: event.id, subscriber: userId};
    registerToEvents(request, token).then(data => {
      setIsLoading(false);
      if (data === SUBSCRIBE_ERROR) {
        setError(true);
      } else {
        setError(false);
      }
      getPublishedEvents().then(result => {
        if (result !== EVENTS_ERROR) {
          dispatch(updateEvents(result));
        }
      });
    });
  };

  const address =
    event.localisation.address +
    ', ' +
    event.localisation.zipCode +
    ' ' +
    event.localisation.city;

  const taken = event.availability.placeTaken;
  const total = event.availability.totalPlaces;
  const availability = t('events.availability', {
    availablePlaces: total - taken,
    totalPlaces: total,
  });
  const startTime = formatHourDate(event.schedule.start);
  const endTime = formatHourDate(event.schedule.end);
  const date = formatDate(event.schedule.end);

  const schedule = t('events.schedule', {start: startTime, end: endTime, date});

  const alreadySubscribe =
    userId && event.subscribers && event.subscribers.find(s => s === userId);
  return (
    <Container>
      <Content>
        <SpecialText
          content={event.title}
          size={22}
          textAlign={'center'}
          color={colors.yellow}
        />
        <Space />
        <SpecialText content={event.description} size={18} />
        <Space />
        <SpecialText content={address} size={14} />
        <Space />
        {token && (
          <Messages>{error && <Error>{t('events.error')}</Error>}</Messages>
        )}
        <SpecialText content={availability} size={14} textAlign={'right'} />
        <SmallSpace />
        <SpecialText content={schedule} size={14} textAlign={'right'} />
        {token && (
          <SubscribeButton>
            <Button
              text={
                alreadySubscribe
                  ? t('events.registered')
                  : t('events.subscribe')
              }
              onPress={() => subscribe()}
              isLoading={isLoading}
              disabled={!!alreadySubscribe}
              color={alreadySubscribe ? colors.green : undefined}
            />
          </SubscribeButton>
        )}
      </Content>
    </Container>
  );
};

export default Event;
