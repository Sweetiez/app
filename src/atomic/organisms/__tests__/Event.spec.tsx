import * as React from 'react';
import renderer from 'react-test-renderer';
import '../../../utils/mocks';
import ReduxProvider from '../../../utils/provider';
import Event from '../Event';
import {store} from '../../../store/configureStore';
import {EventModel} from '../../../model';

describe('Event', () => {
  it('renders correctly', () => {
    const event: EventModel = {
      id: '1',
      title: 'title',
      description: 'descritpion',
      localisation: {
        id: 'id',
        address: 'address',
        zipCode: 'zipCode',
        city: 'city',
      },
      availability: {totalPlaces: 3, placeTaken: 1},
      schedule: {
        start: '12/12/2022',
        end: '12/12/2022',
      },
    };
    const tree = renderer
      .create(
        <ReduxProvider reduxStore={store}>
          <Event event={event} />
        </ReduxProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
