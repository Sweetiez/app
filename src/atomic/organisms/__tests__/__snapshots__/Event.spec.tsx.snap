// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "flexDirection": "row",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
        "shadowColor": "rgba(0, 0, 0, 0.07)",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      color="#dba970"
      size={22}
      style={
        Array [
          Object {
            "color": "#dba970",
            "fontFamily": "Pompiere-Regular",
            "fontSize": 22,
            "textAlign": "center",
            "textDecorationColor": "black",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
        ]
      }
      textAlign="center"
      textDecoration="none"
      textTransform="none"
    >
      title
    </Text>
    <View
      style={
        Array [
          Object {
            "height": 10,
          },
        ]
      }
    />
    <Text
      color="#000000"
      size={18}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Pompiere-Regular",
            "fontSize": 18,
            "textAlign": "left",
            "textDecorationColor": "black",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
        ]
      }
      textAlign="left"
      textDecoration="none"
      textTransform="none"
    >
      descritpion
    </Text>
    <View
      style={
        Array [
          Object {
            "height": 10,
          },
        ]
      }
    />
    <Text
      color="#000000"
      size={14}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Pompiere-Regular",
            "fontSize": 14,
            "textAlign": "left",
            "textDecorationColor": "black",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
        ]
      }
      textAlign="left"
      textDecoration="none"
      textTransform="none"
    >
      address, zipCode city
    </Text>
    <View
      style={
        Array [
          Object {
            "height": 10,
          },
        ]
      }
    />
    <Text
      color="#000000"
      size={14}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Pompiere-Regular",
            "fontSize": 14,
            "textAlign": "right",
            "textDecorationColor": "black",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
        ]
      }
      textAlign="right"
      textDecoration="none"
      textTransform="none"
    >
      events.availability
    </Text>
    <View
      style={
        Array [
          Object {
            "height": 5,
          },
        ]
      }
    />
    <Text
      color="#000000"
      size={14}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Pompiere-Regular",
            "fontSize": 14,
            "textAlign": "right",
            "textDecorationColor": "black",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
        ]
      }
      textAlign="right"
      textDecoration="none"
      textTransform="none"
    >
      events.schedule
    </Text>
  </View>
</View>
`;
